src/test/java/org/opentripplanner/routing/fares/OrcaFareServiceTest.java

```diff
@@ -0,0 +1,162 @@
+        calculateFare(rides, Fare.FareType.electronicYouth, 300f);
+    }
+
+    /**
+     * WSF do not except free transfers. This test is to make sure the rider is charged the cash price for WSF as well
+     * as the highest fare where Orca can be used.
+     */
+    @Test
+    public void calculateFareWithNoFreeTransfer() {
+        List<Ride> rides = Lists.newArrayList();
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 0));
+        rides.add(getRide(WASHINGTON_STATE_FERRIES_AGENCY_ID, 1));
+        rides.add(getRide(COMM_TRANS_AGENCY_ID, 2));
+        calculateFare(rides, Fare.FareType.regular, DEFAULT_RIDE_PRICE_IN_CENTS * 3);
+        calculateFare(rides, Fare.FareType.senior, DEFAULT_RIDE_PRICE_IN_CENTS * 3);
+        calculateFare(rides, Fare.FareType.youth, DEFAULT_RIDE_PRICE_IN_CENTS * 3);
+        calculateFare(rides, Fare.FareType.electronicSpecial, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 125f);
+        calculateFare(rides, Fare.FareType.electronicRegular, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + DEFAULT_RIDE_PRICE_IN_CENTS);
+        calculateFare(rides, Fare.FareType.electronicSenior, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 125f);
+        calculateFare(rides, Fare.FareType.electronicYouth, 200f + DEFAULT_RIDE_PRICE_IN_CENTS + 0f);
+    }
+
+    /**
+     * Total trip time is 2h 30m. The first four transfers are within the permitted two hour window. A single (highest)
+     * Orca fare will be charged for these transfers. The fifth transfer is outside of the two hour window and will be
+     * charged a cash rate. At this point, the two hour window will start again so the final transfer will be
+     * charged at a discount rate... if using Orca.
+     */
+    @Test
+    public void calculateFareThatExceedsTwoHourFreeTransferWindow() {
+        List<Ride> rides = Lists.newArrayList();
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 0));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 30));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 60));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 90));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 120));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 150));
+
+        calculateFare(rides, Fare.FareType.regular, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.senior, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.youth, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.electronicSpecial, 100f + 0f + 0f + 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 100f);
+        calculateFare(rides, Fare.FareType.electronicRegular, DEFAULT_RIDE_PRICE_IN_CENTS + 0f + 0f + 0f +
+            DEFAULT_RIDE_PRICE_IN_CENTS + DEFAULT_RIDE_PRICE_IN_CENTS);
+        calculateFare(rides, Fare.FareType.electronicSenior, 100f + 0f + 0f + 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 100f);
+        calculateFare(rides, Fare.FareType.electronicYouth, 200f + 0f + 0f + 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 200f);
```

Comment:
I'm pretty sure the youth reduced fare for Kitsap is $1. See [here](https://www.kitsaptransit.com/fares/fares).